docs/docsite/rst/dev_guide/platforms/vmware_guidelines.rst

```diff
@@ -0,0 +1,76 @@
+.. _VMware_module_development:
+
+***********************************************
+Guidelines for VMware module module development
+***********************************************
+
+The VWware modules and these guidelines are maintained by the VMware Working Group.  For
+further information the team community page: `https://github.com/ansible/community/wiki/VMware`.
+
+.. contents::
+   :local:
+
+Testing
+=======
+
+Most of the existing modules are covered by functional tests. The tests are located in the :file:`test/integration/targets/`.
+
+Test environment
+----------------
+
+By default, the tests are run against a vcenter API simulator called `vcsim <https://github.com/vmware/govmomi/tree/master/vcsim>`. ``ansible-test`` will automatically pull a vcsim container and use it to set-up the test environment.
+
+You can trigger the test of a module manually with the ``ansible-test`` command. For example, to trigger ``vcenter_folder`` tests:
+
+.. code-block:: shell
+
+    source hacking/env-setup
+    ansible-test integration --python 3.7 vcenter_folder
+
+You will need to install a couple of Python dependencies, like pyVim, pyvmomi and requests.
+
+``vcsim`` is handy because it's much more fast that than a regular test environment. However, it does not
+support all the ESXi or vcenter features.
+
+When `PR #55772 <https://github.com/ansible/ansible/pull/55772>` and `PR #56605 <https://github.com/ansible/ansible/pull/56605>` will be merged, you will also be able
+to target a regular VMware environment.
+For this purpose, you need to prepare a configuration file that discribe your set-up. The file
+should be called :file:`test/integration/cloud-config-vcenter.ini` and based on
+:file:`test/integration/cloud-config-vcenter.ini.template`.
+
+In addition to this file, you may need to adjust to match the configuration of your lab
+:file:`test/integration/targets/prepare_vmware_tests/vars/real_lab.yml`.
+
+Once your configuration is ready, you can trigger a run with the following command:
+
+.. code-block:: shell
+
+    source hacking/env-setup
+    env VMWARE_TEST_PLATFORM=worldstream ansible-test integration --python 3.7 vmware_host_firewall_manager
+
+convention and best practice
+============================
+
+datacenter argument with ESXi
+-----------------------------
+
+``datacenter`` parameter should not use ``ha-datacenter`` by default. This because the user may
+not realize that Ansible silently target the wrong datacenter.
+
+esxi_hostname should not be mandatory
+-------------------------------------
+
+Some modules can seamlessly works with a vcenter or an ESXi.
```

Comment:
```suggestion
Some modules can seamlessly works with a vCenter or an ESXi (Depending upon the functionality provided by ESXi or vCenter)
```

Reply:
I rephrase to: Depending upon the functionality provided by ESXi or vCenter, some modules can seamlessly work with both.